components/script/dom/document.rs

```diff
@@ -2837,7 +2837,12 @@ impl Document {
         let trusted_pending = Trusted::new(pending);
         let trusted_promise = TrustedPromise::new(promise.clone());
         let handler = ElementPerformFullscreenEnter::new(trusted_pending, trusted_promise, error);
-        let script_msg = CommonScriptMsg::Task(ScriptThreadEventCategory::EnterFullscreen, handler, pipeline_id);
+        let script_msg = CommonScriptMsg::Task(
+            ScriptThreadEventCategory::EnterFullscreen,
+            handler,
+            pipeline_id,
+            TaskSourceName::DOMManipulation,
```

Comment:
Please add a NOTE here that the spec mentions running these steps "in parallel", not necessarily as a queued task. For now using the `DOMManipulation` seems OK however. See Step 7 of https://fullscreen.spec.whatwg.org/#dom-element-requestfullscreen

Reply:
I've filed a bit more of an explanation at https://github.com/servo/servo/issues/21600